Adamant/Modules/Wallets/TransactionsListViewControllerBase.swift

```diff
@@ -134,18 +138,31 @@ class TransactionsListViewControllerBase: UIViewController {
         let nib = UINib.init(nibName: "TransactionTableViewCell", bundle: nil)
         tableView.register(nib, forCellReuseIdentifier: cellIdentifierFull)
         tableView.register(nib, forCellReuseIdentifier: cellIdentifierCompact)
-        tableView.dataSource = self
         tableView.delegate = self
         tableView.refreshControl = refreshControl
         tableView.tableHeaderView = UIView()
     }
     
     @MainActor
-    func update(_ transactions: [CoinTransaction]) {
-        self.transactions = transactions.sorted(by: {
-            (($0.date ?? NSDate()) as Date) > (($1.date ?? NSDate()) as Date)
-        })
-        self.tableView.reloadData()
+    func update(_ transactions: [TransactionDetails]) {
+        let transactions = transactions.map {
+            SimpleTransactionDetails($0)
+        }
+        
+        update(transactions)
+    }
+    
+    @MainActor
+    func update(_ transactions: [SimpleTransactionDetails]) {
```

Comment:
To avoid extra updates use HashableWrapper: https://github.com/Adamant-im/adamant-iOS/pull/346/files#diff-31edf532106634e7fe33d62e50063e9eb0bfa3eb7a46d72118e2b6bd55eed784R8

And change it so it will use as id just string.